File: PRO/T11/AcedamiaDB/src/main/java/model/Alumno.java
package model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Alumno {

    private int id;
    private String nombre, apellido, correo;
    private int telefono;

    public Alumno(String nombre, String apellido, String correo, int telefono) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.correo = correo;
        this.telefono = telefono;
    }

    public Alumno(String nombre, String apellido, int telefono) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.telefono = telefono;
    }

    public void mostrarDatos(){
        System.out.println("id = " + id);
        System.out.println("nombre = " + nombre);
        System.out.println("apellido = " + apellido);
        System.out.println("correo = " + correo);
        System.out.println("telefono = " + telefono);
    }

    public void agregarAlumno(){} //?????
    public void agregarProfesor(){} // ??????
}

File: PMDM/curso2526/T1/AppInicial/src/model/Alumno.kt
package model

// constructores primarios: tiene todas las variables obligatorias para el objeto
open class Alumno (var id: Int=0, var nombre: String, var apellido: String){

    var correo: String? = null
    // constructores secundarios: aquellos constructores que SE BASAN EN EL PRIMARIO mas lo adicional
    constructor(id: Int, nombre: String, apellido: String, correo: String): this(id,nombre,apellido) {
        this.correo = correo;
    }

    open fun mostrarDatos(){
        println("id = ${id}")
        println("nombre = ${nombre}")
        println("apellido = ${apellido}")
        println("correo = ${correo ?: "sin correo"}")
        println("Tu correo tiene ${correo?.length ?: "0"}")
    }
}